packages/devtools_app/lib/src/extensions/extension_settings.dart

```diff
@@ -0,0 +1,204 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:devtools_shared/devtools_extensions.dart';
+import 'package:flutter/material.dart';
+
+import '../shared/analytics/analytics.dart' as ga;
+import '../shared/analytics/constants.dart' as gac;
+import '../shared/common_widgets.dart';
+import '../shared/dialogs.dart';
+import '../shared/globals.dart';
+import '../shared/theme.dart';
+import '../shared/utils.dart';
+
+/// A [ScaffoldAction] that, when clicked, will open a dialog menu for
+/// managing DevTools extension states.
+class ExtensionSettingsAction extends ScaffoldAction {
+  ExtensionSettingsAction({super.key, Color? color})
+      : super(
+          icon: Icons.extension_rounded,
+          tooltip: 'DevTools Extensions',
+          color: color,
+          onPressed: (context) {
+            unawaited(
+              showDialog(
+                context: context,
+                builder: (context) => const ExtensionSettingsDialog(),
+              ),
+            );
+          },
+        );
+}
+
+class ExtensionSettingsDialog extends StatelessWidget {
+  const ExtensionSettingsDialog({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final availableExtensions = extensionService.availableExtensions.value;
+    final dialogHeight = scaleByFontFactor(300.0);
```

Comment:
yes this is arbitrary. Other dialogs are wrapped to the size of their content, but since we have a scrollable in this dialog, we need to git it a fixed height. Added a comment.